lib/active_storage_support/base64_attach.rb

```diff
@@ -5,13 +5,18 @@ module Base64Attach
 
     def attachment_from_data(attachment)
       if attachment.is_a?(Hash)
-        base64_data = attachment.delete(:data)
-        if base64_data.try(:is_a?, String) && base64_data =~ /^data:(.*?);(.*?),(.*)$/
+        attachment.symbolize_keys!
+        attachment = fill_attachment_data(attachment, attachment.delete(:data))
+      end
+
+      attachment
+    end
 
-          attachment[:io] = StringIO.new(Base64.decode64(Regexp.last_match(3)))
-          attachment[:content_type] = Regexp.last_match(1) unless attachment[:content_type]
-          attachment[:filename] = Time.current.to_i.to_s unless attachment[:filename]
-        end
+    def fill_attachment_data(attachment, base64_data)
```

Comment:
In codeclimate I see 
![image](https://user-images.githubusercontent.com/6373536/50691476-9614a500-100f-11e9-9007-37b08c28d087.png)

what do you think about refactoring this a bit more to get an `A`?

Reply:
Totally agree. Talking about CodeClimate, we could add test coverage badge as well. We have it in active_admin_chat https://github.com/rootstrap/active_admin_chat